[CI/Build] pre-commit markdown-lint fails on main: SECURITY.md violates MD004 and MD007

## Description

The **Run pre-commit hooks check file lint** job fails on `main` and on every open pull request. `SECURITY.md`, added by #2651 (merge commit `a7788256`), mixes unordered-list markers and uses four-space nested list indentation, which violates the repository's own markdownlint configuration (`MD004/ul-style`, `MD007/ul-indent`). Because the `markdown-lint` target globs all `**/*.md`, the failure is not limited to PRs that touch the file — any PR inherits it, since GitHub lints the merge with `main`. Observed on `main` @ `a7788256`, and reproduced locally on darwin/arm64 (markdownlint-cli via Homebrew).

## Steps to Reproduce

1. Fresh checkout of `main` (`a7788256` or later).
2. Run the same target CI runs:
   ```
   make markdown-lint
   ```
   or invoke the linter directly on the file:
   ```
   markdownlint -c tools/linter/markdown/markdownlint.yaml SECURITY.md
   ```
3. The command exits non-zero with 12 violations.

## Error / Evidence

```
SECURITY.md:48:1 MD004/ul-style Unordered list style [Expected: dash; Actual: asterisk]
SECURITY.md:49:1 MD004/ul-style Unordered list style [Expected: dash; Actual: asterisk]
SECURITY.md:57:1 MD004/ul-style Unordered list style [Expected: dash; Actual: asterisk]
SECURITY.md:59:1 MD004/ul-style Unordered list style [Expected: dash; Actual: asterisk]
SECURITY.md:61:1 MD004/ul-style Unordered list style [Expected: dash; Actual: asterisk]
SECURITY.md:62:5 MD004/ul-style Unordered list style [Expected: dash; Actual: asterisk]
SECURITY.md:62:1 MD007/ul-indent Unordered list indentation [Expected: 2; Actual: 4]
SECURITY.md:63:5 MD004/ul-style Unordered list style [Expected: dash; Actual: asterisk]
SECURITY.md:63:1 MD007/ul-indent Unordered list indentation [Expected: 2; Actual: 4]
SECURITY.md:64:5 MD004/ul-style Unordered list style [Expected: dash; Actual: asterisk]
SECURITY.md:64:1 MD007/ul-indent Unordered list indentation [Expected: 2; Actual: 4]
SECURITY.md:66:1 MD004/ul-style Unordered list style [Expected: dash; Actual: asterisk]
make[3]: *** [tools/make/linter.mk:13: markdown-lint] Error 1
make[1]: *** [tools/make/pre-commit.mk:19: precommit-check] Error 1
```

Failing CI job on an unrelated PR, demonstrating the blast radius:
https://github.com/vllm-project/semantic-router/actions/runs/30040983518/job/89320627905

Regression timeline for the pre-commit job on `main`:

| Commit | Result |
|---|---|
| `8fe6b342` | pass |
| `1222d205` | pass |
| `a7788256` (merge of #2651, adds `SECURITY.md`) | **fail** |
| every commit since | fail |

## Code Path

- `SECURITY.md:48-66` — list block mixing `*` markers with the `-` markers used earlier in the same file; `MD004` is configured as `consistent`, so the first marker in the file sets the expectation.
- `SECURITY.md:62-64` — nested list items indented four spaces; `MD007` expects two.
- `tools/make/linter.mk:11` — `markdown-lint` target; globs `**/*.md`, so a violation in any non-ignored file fails the whole job.
- `tools/linter/markdown/markdownlint.yaml` — the ruleset the file must satisfy.
- `tools/make/pre-commit.mk:19` — `precommit-check` propagates the failure to the CI job.

## Expected vs Actual Behavior

- **Expected:** `make markdown-lint` passes on `main`, so the pre-commit job is green and PRs only fail for problems they actually introduce.
- **Actual:** the target exits 1 on `SECURITY.md`, turning the pre-commit job red on `main` and on every open PR regardless of its contents.

## Proposed Fix

Run `markdownlint --fix` on `SECURITY.md` (or `make markdown-lint-fix`, `tools/make/linter.mk:24`). The correction is purely stylistic — list markers `*` → `-` and nested indentation 4 → 2 spaces — with no wording, link, or policy changes. Verified afterwards by linting every tracked markdown file:

```
$ git ls-files '*.md' | xargs markdownlint -c tools/linter/markdown/markdownlint.yaml
(no output — 905 files clean)
```

--- a/SECURITY.md
+++ b/SECURITY.md
@@ -45,8 +45,8 @@ Minor issues such as informational disclosures, logging errors, non-exploitable 
 
 When a security report is accepted, the fix process depends on the severity:
 
-* **CRITICAL and HIGH severity**: Fixes are developed in a private security fork and coordinated with the prenotification group before public disclosure.
-* **MODERATE and LOW severity**: Fixes are developed and submitted as public pull requests. These issues do not require embargo since they do not enable arbitrary code execution or significant data breach, and public visibility accelerates community review and adoption of the fix.
+- **CRITICAL and HIGH severity**: Fixes are developed in a private security fork and coordinated with the prenotification group before public disclosure.
+- **MODERATE and LOW severity**: Fixes are developed and submitted as public pull requests. These issues do not require embargo since they do not enable arbitrary code execution or significant data breach, and public visibility accelerates community review and adoption of the fix.
 
 The vLLM Semantic Router maintainer team reserves the right to adjust the disclosure approach on a case-by-case basis, taking into account factors such as active exploitation, unusual attack surface, or coordination requirements with vLLM or downstream vendors.
 
@@ -54,13 +54,13 @@ The vLLM Semantic Router maintainer team reserves the right to adjust the disclo
 
 For certain security issues of CRITICAL, HIGH, or MODERATE severity level, we may prenotify the vLLM vulnerability management team and certain organizations or vendors that ship vLLM Semantic Router. The purpose of this prenotification is to allow for a coordinated release of fixes for severe issues.
 
-* This prenotification will be in the form of a private email notification. It may also include adding security contacts to the GitHub security advisory, typically a few days before release.
+- This prenotification will be in the form of a private email notification. It may also include adding security contacts to the GitHub security advisory, typically a few days before release.
 
-* If you wish to be added to the prenotification group, please contact the vLLM Semantic Router maintainer team and copy the members of the [vLLM vulnerability management team](https://docs.vllm.ai/en/latest/contributing/vulnerability_management.html). Each vendor contact will be analyzed on a case-by-case basis.
+- If you wish to be added to the prenotification group, please contact the vLLM Semantic Router maintainer team and copy the members of the [vLLM vulnerability management team](https://docs.vllm.ai/en/latest/contributing/vulnerability_management.html). Each vendor contact will be analyzed on a case-by-case basis.
 
-* Organizations and vendors who either ship or use vLLM Semantic Router are eligible to join the prenotification group if they meet at least one of the following qualifications:
-    * Substantial internal deployment leveraging the upstream vLLM Semantic Router project.
-    * Established internal security teams and comprehensive compliance measures.
-    * Active and consistent contributions to the upstream vLLM Semantic Router project.
+- Organizations and vendors who either ship or use vLLM Semantic Router are eligible to join the prenotification group if they meet at least one of the following qualifications:
+  - Substantial internal deployment leveraging the upstream vLLM Semantic Router project.
+  - Established internal security teams and comprehensive compliance measures.
+  - Active and consistent contributions to the upstream vLLM Semantic Router project.
 
-* We may withdraw organizations from receiving future prenotifications if they release fixes or any other information about issues before they are public. Group membership may also change based on policy refinements for who may be included.
+- We may withdraw organizations from receiving future prenotifications if they release fixes or any other information about issues before they are public. Group membership may also change based on policy refinements for who may be included.

